### DISCUSSION TITLE
Slow requests due to possibly incorrect break.

### DISCUSSION THREAD

**Discussion by blubard** [encode/starlette#1868] [Potential Issue] [answered]
With FastAPI, under certain circumstances I was noticing some calls were very slow.

It turned out that in starlette version 0.20.1, a [break](https://github.com/encode/starlette/compare/0.20.0...0.20.1#diff-931c53548e8908544bd0818ee1a8d5f1ada4f07430a439e9322ab369eb40a8acR58) was added into the `body_stream` method (`BaseHTTPMiddleware`).

This break omits the final `__anext__` of MemoryObjectReceiveStream (resp. UnreliableObjectReceiveStream) and hence will never raise the necessary `StopAsyncIteration` to properly exit the async for loop.

I am not sure why this results in a "slow" response and not in something more fatal, I guess my knowledge of async python is too limited... (is there something timing out?). Also I am not sure why this occurs not for all my calls; after I raise in fastapi, the follow up REST request is fast again.

A "fix" was to close the `recv_stream` instead of using the break.

Is this a real issue or was this break intended to behave as it does? Is it likely that the issue is due to invalid usage from my side?

**Comment by Kludex** [comment, marked as answer]
Yeah, this is correct. I didn't see it soon enough.

That logic was removed. Thanks :+1:
